### PR TITLE
dcp-855 error handling when accession is not found

### PR DESCRIPTION
**Pull request by amnonkhen**
dcp-855

### CODE REVIEW

**Review by ESapenaVentura — CHANGES_REQUESTED** [review]
small comment on config class

File: geo_to_hca/config.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def _parse_bool(val: Union[str, bool]) -> bool:  # pylint: disable=E1136
+    return val if type(val) == bool else val.lower() in ['true', 'yes', '1']
+
+
+class Config:
+    IS_INTERACTIVE: bool = 'true'
+    DEBUG: bool = 'false' \
+                  ''
```
[the remark this answers is not in the document]
Comment (amnonkhen) [reply geo_to_hca/config.py:16]: 1. `Config` can accept strings as well as booleans.
2. a careless keypress, probably.

File: geo_to_hca/utils/sra_utils.py [hunk 2]
```diff
@@ -98,7 +98,11 @@ def get_srp_metadata(srp_accession: str) -> pd.DataFrame:
                                  retmode="text",
                                  mode='prepare')
     log.debug(f'srp_metadata url: {efetch_request.url}')
-    return pd.read_csv(efetch_request.url)
+    srp_metadata = pd.read_csv(efetch_request.url)
+    if 'Run' not in srp_metadata.columns:
+        raise RuntimeError(f'cannot build the srp_metadata from {efetch_request.url}: '
```
Comment (MightyAx) [inline geo_to_hca/utils/sra_utils.py:103]: Should we throw a more specific exception here?

File: geo_to_hca/utils/sra_utils.py [hunk 3]
```diff
@@ -130,8 +134,10 @@ def request_accession_info(accessions: [], accession_type: str) -> object:
     """
     if accession_type == 'biosample':
         db = f'biosample'
-    if accession_type == 'experiment':
+    elif accession_type == 'experiment':
         db = f'sra'
+    else:
+        raise ValueError(f'unsupported accession_type: {accession_type}')
```
Comment (MightyAx) [inline geo_to_hca/utils/sra_utils.py:140]: Should we throw a more specific exception here?

**Pull request merged**